coalib/core/Core.py

```diff
@@ -183,6 +186,91 @@ def finish_task(bear,
                     exc_info=ex)
 
 
+def initialize_dependencies(bears):
+    """
+    Initializes and returns a ``DependencyTracker`` instance together with a
+    set of bears ready for scheduling.
+
+    This function acquires, processes and registers bear dependencies
+    accordingly using a consumer-based system, where each dependency bear has
+    only a single instance per section and file-dictionary.
+
+    The bears set returned accounts for bears that have dependencies and
+    excludes them accordingly. Dependency bears that have themselves no further
+    dependencies are included so the dependency chain can be processed
+    correctly.
+
+    :param bears:
+        The set of bears to run that serve as an entry-point.
+    :return:
+        A tuple with ``(dependency_tracker, bears_to_schedule)``.
+    """
+    # Pre-collect bears in a set as we use them more than once. Especially
+    # remove duplicate instances.
+    bears = set(bears)
+
+    dependency_tracker = DependencyTracker()
+
+    # For a consumer-based system, we have a situation which can be visualized
+    # with a graph:
+    #
+    # (section1, file_dict1) (section1, file_dict2) (section2, file_dict2)
```

Comment:
As sections and filedicts are feeded in separately this case is theoretically possible (if someone invokes the core directly).
But actually we could merge filedicts that refer to the same section. What do you think @Udayan12167 ?